flow/Arena.h

```diff
@@ -998,7 +1003,351 @@ class VectorRef : public ComposedIdentifier<T, 0x8>, public VectorRefPreserializ
 	}
 };
 
-template<class T>
+// This is a VectorRef that optimizes for tiny to small sizes.
+// It keeps the first #InlineMembers on the stack - which means
+// that all of them are always copied. This should be faster
+// when you expect the vector to be usually very small as it
+// won't need allocations in these cases.
+template <class T, int InlineMembers = 1>
+class SmallVectorRef {
+public:
+	// types
+	template <bool isConst>
+	class iterator_impl {
+		using self_t = iterator_impl<isConst>;
+		using VecType = SmallVectorRef<T, InlineMembers>;
+		std::conditional_t<isConst, const VecType*, VecType*> vec = nullptr;
+		int idx = 0;
+
+	public:
+		using Category = std::random_access_iterator_tag;
+		using value_type = std::conditional_t<isConst, const T, T>;
+		using difference_type = int;
+		using pointer = value_type*;
+		using reference = value_type&;
+		friend class SmallVectorRef<T, InlineMembers>;
+		template<bool I>
```

Comment:
These functions only get defined for `I == isConst`, so the class `I` isn't necessary.

Reply:
I though the same thing, but without it the compiler complains... I think it is in fact necessary, because when this is getting defined (as free function), `IsConst` is not available anymore.